fix #2407 by tslinting tests before running

anurse opened the pull request:
I also saw some extra logging that we intend to suppress and ~~noticed a weird compilation error (might not be happening on CI because of a compiler version difference or something, but it's benign to fix)~~ nothing else. [Everything is fine. We're all fine here. How are you?](https://www.youtube.com/watch?v=KYAbFqkvzQA)

anurse commented on clients/ts/signalr/tests/HttpConnection.test.ts:
```diff
@@ -351,24 +351,24 @@ describe("HttpConnection", () => {
-                return new HttpResponse(204, "No Content", "");
-            })
-            .on("DELETE", (r) => new HttpResponse(202));
+            .on("POST", /negotiate$/, () => {
```
TSLint doesn't like unused parameters.

anurse commented on clients/ts/signalr/tests/HttpConnection.test.ts:
```diff
@@ -644,16 +644,16 @@ describe("HttpConnection", () => {
                 ...commonOptions,
-                EventSource: customEventSourceType as EventSourceConstructor,
-                httpClient: new TestHttpClient().on("POST", (r) => {
+                EventSource: customEventSourceType as any,
```
TS compiler was failing because our custom EventSource type defined in this test doesn't actually implement all the things in EventSource. I guess casting doesn't really satisfy it, but casting to `any` is basically like telling the TS compiler to leave us alone.

anurse commented on clients/ts/signalr/tests/HttpConnection.test.ts:
```diff
@@ -674,15 +676,17 @@ describe("HttpConnection", () => {
             const options: IHttpConnectionOptions = {
                 ...commonOptions,
-                WebSocket: customWebSocketType as WebSocketConstructor,
+                WebSocket: TestWebSocket,
```
Not sure why, but this doesn't need the `as` cast.

anurse commented on clients/ts/signalr/tests/HttpConnection.test.ts:
```diff
@@ -643,17 +643,19 @@ describe("HttpConnection", () => {
 
-            const customEventSourceType = class EventSource {
-                constructor(url: string, eventSourceInitDict?: EventSourceInit) {
+            class TestEventSource {
```
These were mostly a style change rather than the linter. Since the class can just be local to the function and have it's own name, there's no need to assign it to a variable.

BrennanConroy commented:
You forgot "no-floating-promises"

BrennanConroy commented on clients/ts/signalr/tests/HttpConnection.test.ts:
```diff
@@ -643,17 +643,19 @@ describe("HttpConnection", () => {
+            class TestEventSource {
+                // The "_" prefix tell TypeScript not to worry about unused parameter, but tslint doesn't like it.
+                // tslint:disable-next-line:variable-name
+                constructor(_url: string, _eventSourceInitDict: EventSourceInit) {
```
`"no-unused-variable": [true, {"ignore-pattern": "^_"}]`?

anurse commented:
:up: :date: added `no-floating-promises` and fixed the things. The only product changes were places we were intentionally ignoring the Promise return value (because we had no need to `await` it).
